### PR TITLE
Improve Ember Style Guide

### PR DESCRIPTION
**Pull request by jordanyee**
This PR takes the first step in bringing the Ember Style Guide up to the level of the Ruby and Rails Style Guides.

It also clarifies some of the examples with thanks to @runspired, @AlexBlom, and @anulman for their help.

### CODE REVIEW

File: ember-style-guides.md [hunk 1]
```diff
@@ -1,269 +1,312 @@
...
-Component,
-inject
+  Component,
+  moment: { add }
 } = Ember;
 
-const { service } = inject;
+export default Component.extend({
+  session: service();
+});
```
Comment (jordanyee) [inline ember-style-guides.md:80]: I stubbed this bad example here but there's probably a better or more common anti-pattern we could cite. (You'll want to view the full changes for context.)

File: ember-style-guides.md [hunk 2]
```diff
@@ -1,269 +1,312 @@
...
 
-export default Component.extend();
+  // lifecycle hooks in order of execution
+  model() {},
+  setupController() {},
+
+  // custom functions
+
+  // actions separated by a newline
+});
```
Comment (jordanyee) [inline ember-style-guides.md:108]: This structure example needs examples where there are just comments. It'd be nice to have a common app context throughout the guide. In models we use a blog post. (You'll want to view the full changes for context.)

File: ember-style-guides.md [hunk 3]
```diff
@@ -1,269 +1,312 @@
...
+    },
 
-export default Controller.extend({
-attrs: {}
+    doBar() {
+      // body omitted
+    }
+  }
+});
+```
```
Comment (jordanyee) [inline ember-style-guides.md:209]: The same goes for the component structure example. (You'll want to view the full changes for context.)

File: ember-style-guides.md [hunk 4]
```diff
@@ -1,269 +1,312 @@
...
-  },
+  // computed properties
+  fancyTitle: computed('title', function() {
+    // return something
+  });
 
-  setupController(controller, model) {
-    controller.set(‘attrs’, model);
-  }   
+  // custom functions
```
Comment (jordanyee) [inline ember-style-guides.md:252]: The model structure example is not too bad. Just need the service declaration example again and the custom functions. (You'll want to view the full changes for context.)

**Comment by jordanyee** [comment]
I'd also like to see this guide move to it's own repo like the [Ruby Style Guide](https://github.com/bbatsov/ruby-style-guide). Having it buried behind the current readme seems like an extra step we don't need. Thoughts on https://github.com/isleofcode/ember-style-guide?

**Comment by jordanyee** [comment]
I'd also love to see an approach to animation added to the guide at some point. Right now it's stuffed under miscellaneous but that's for another PR.

**Comment by anulman** [comment]
🚀

**Pull request merged**